addon/index.js

```diff
@@ -130,28 +148,35 @@ function formatInstallData(install, addon) {
 
 AddonManager.addAddonListener({
   onUninstalling: function(addon) {
-    app.send('addons-uninstalling', {
-      id: addon.id,
-      name: addon.name
-    });
+    if (isIdeatownAddon(addon.id)) {
+      app.send('addon-uninstalling', {
+        id: addon.id,
+        name: addon.name
+      });
+    }
   },
   onUninstalled: function(addon) {
-    app.send('addons-uninstalled', {
-      id: addon.id,
-      name: addon.name
-    }, addon);
+    if (isIdeatownAddon(addon.id)) {
+      app.send('addon-uninstalled', {
+        id: addon.id,
+        name: addon.name
+      }, addon);
 
-    store.experiments = store.experiments.filter(ex => ex.id !== addon.id);
-    state.updates = require('exclude')(updates, store.experiments);
+      store.experiments = store.experiments.filter(ex => ex.id !== addon.id);
```

Comment:
You know, it might actually be really nice to make experiments not be an array containing objects, but an object whose keys are the ids of the experiments.

Then, instead of having to iterate an array to find out if something exists, you could just check via `key in obj`. And, instead of having to iterate an array to remove something, you could just delete the key. ¯\_(ツ)_/¯